### PR TITLE
HPCC-16237 Support OR filter in getWorkUnitsSorted()

### PR DESCRIPTION
**Pull request by wangkx**
Also add the code to wutool.cpp for selftests.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by wangkx** [comment]
This PR replaces the old PR https://github.com/hpcc-systems/HPCC-Platform/pull/9726.

I noticed that wutoolTest has 5 failed for Cassandra. Is that because the new tests for the OR filter has not supported  in Cassandra?

@jakesmith @richardkchapman please review.

File: tools/wutool/wutool.cpp [hunk 1]
```diff
@@ -1570,6 +1571,81 @@ class WuTool : public CppUnit::TestFixture
...
+                ASSERT((wu.getState()==WUStateCompleted) || (wu.getState()==WUStateFailed));
+                if ((wu.getState()==WUStateCompleted))
+                    numCompleted++;
+                else
+                    numFailed++;
+            }
+            DBGLOG("%d completed workunits listed and %d failed workunits listed in %d ms", numCompleted, numFailed, msTick()-start);
+        }
+
+        if ((numCompleted > 0) || (numFailed > 0))
```
Comment (richardkchapman) [inline tools/wutool/wutool.cpp]: Not sure what you are trying to achieve here - the tests in wutool should assume that the repository being tested is in a known state (at least as far as the fake workunits created by this test are concerned).

Comment (wangkx) [inline tools/wutool/wutool.cpp]: You are correct and I should remove those two lines. My original thought assumes that a user will read the test result by line 1602). If the existing system does have both completed WUs and failed WUs, the user should know whether the numCompleted and numFailed are correct not from that result. In fact, I should not assumes that a user will read the test result.

**Comment by wangkx** [comment]
@richardkchapman please review my changes.

**Comment by richardkchapman** [comment]
See also #9801

**Comment by richardkchapman** [comment]
Superseded by #9801

**Pull request closed without merging**